src/python/dxpy/bindings/dxfile.py

```diff
@@ -112,24 +120,29 @@ def _ensure_http_threadpool(cls):
         if cls._http_threadpool is None:
             cls._http_threadpool = dxpy.utils.get_futures_threadpool(max_workers=cls._http_threadpool_size)
 
+    @classmethod
+    def _set_file_limits(cls, file_limits):
+        cls._buffer_size = file_limits['minimumPartSize']
+        cls._maximum_parts = file_limits['maximumNumParts']
+        cls._minimum_part_size = file_limits['minimumPartSize']
+        cls._maximum_part_size = file_limits['maximumPartSize']
+
     def __init__(self, dxid=None, project=None, mode=None,
-                 read_buffer_size=DEFAULT_BUFFER_SIZE, write_buffer_size=DEFAULT_BUFFER_SIZE):
+                 read_buffer_size=DEFAULT_BUFFER_SIZE, write_buffer_size=DEFAULT_BUFFER_SIZE, file_size=1, file_is_mmapd=False):
```

Comment:
file_size should default to None, or some other special value, since clients are not obligated to supply it. Also, it's not actually binding on the client to conform to this, so consider calling this expected_file_size.